elastic/kibana
# [Response Ops][Alerting] Fixing bug with using `runSoon` on pre-8.x rule

Resolves https://github.com/elastic/kibana/issues/142293

## Summary

Updates `runSoon` function to use scheduled task id to run rule instead of rule ID. Using the rule ID causes issues for pre-8.x rules where the scheduled task ID was randomly generated and did not match the rule ID. Also added some additional try/catches to provide more graceful error handling in the case of missing tasks.

## To Verify
1. Create a rule in a pre-8.x branch
2. Using the same ES data, run this branch. Navigate to the Rules UI and select `Run Rule` from the menu dropdown. This should return a success toast saying that the rule has been scheduled to run.

### Checklist
